lib/pages/surveydetail/survey_detail_page.dart

```diff
@@ -1,21 +1,65 @@
 import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:flutter_survey/di/di.dart';
 import 'package:flutter_survey/extensions/build_context_ext.dart';
+import 'package:flutter_survey/pages/surveydetail/survey_detail_state.dart';
+import 'package:flutter_survey/pages/surveydetail/survey_detail_view_model.dart';
 import 'package:flutter_survey/pages/surveydetail/widget/start_survey.dart';
 import 'package:flutter_survey/pages/uimodel/survey_ui_model.dart';
+import 'package:flutter_survey/usecase/get_survey_detail_use_case.dart';
+
+final surveyDetailViewModelProvider =
+    StateNotifierProvider.autoDispose<SurveyDetailViewModel, SurveyDetailState>(
+        (ref) {
+  return SurveyDetailViewModel(getIt.get<GetSurveyDetailUseCase>());
+});
+
+final _surveyStreamProvider = StreamProvider.autoDispose<SurveyUiModel>(
+    (ref) => ref.watch(surveyDetailViewModelProvider.notifier).surveyStream);
+
+class SurveyDetailPage extends ConsumerStatefulWidget {
+  @override
+  _SurveyDetailPageState createState() {
+    return new _SurveyDetailPageState();
+  }
+}
+
+class _SurveyDetailPageState extends ConsumerState<SurveyDetailPage> {
+  @override
+  void initState() {
+    super.initState();
+    Future.delayed(Duration.zero, () {
+      final survey = context.arguments as SurveyUiModel;
+      ref.read(surveyDetailViewModelProvider.notifier).loadSurveyDetail(survey);
+    });
+  }
 
-class SurveyDetailPage extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
-    final survey = context.arguments as SurveyUiModel;
-    return _buildSurveyPage(survey);
+    final uiModel = ref.watch(_surveyStreamProvider).value;
+    return Scaffold(
+      body: ref.watch(surveyDetailViewModelProvider).when(
+            init: () => const SizedBox.shrink(),
+            loading: () => const SizedBox.shrink(),
+            success: () => _buildSurveyPage(uiModel),
+            error: (message) {
+              WidgetsBinding.instance?.addPostFrameCallback((_) {
+                ScaffoldMessenger.of(context).showSnackBar(SnackBar(
+                    content:
+                        Text(message ?? context.localization.errorGeneric)));
+              });
+              return _buildSurveyPage(uiModel);
+            },
+          ),
+    );
   }
 
-  Widget _buildSurveyPage(SurveyUiModel survey) {
+  Widget _buildSurveyPage(SurveyUiModel? survey) {
     return Scaffold(
       resizeToAvoidBottomInset: false,
-      body: StartSurvey(
-        survey: survey,
-      ),
+      body: survey != null
+          ? StartSurvey(survey: survey)
+          : const SizedBox.shrink(),
```

Comment:
@luongvo we can get rid of Scaffold here since we include it outside 👌

Reply:
@manh-t oops, notes this to fix it in PRs next milestones :pray:

Reply:
@manh-t I fixed this issue here https://github.com/luongvo/flutter-survey/pull/64/commits/bb0aea19d0f28fc1bd5336b1e0923d698866424c#diff-92a95602b8f6dafa4363851002551ef584951471cc52a1690b14999ba9adbdc1L69 yesterday 👌